### PR TITLE
Using existing tables

### PR DESCRIPTION
**Pull request by rowild**
A new document in the documentation that deals with recommendations and culprits of using tables of an existing project.
WIP!

### CODE REVIEW

**Comment by rowild** [comment]
@phazonoverload Hi! I started a new document dealing with what is mentioned above. Currently set to draft. I was wondering if you had a moment and give me some initial feedback on documentation-related things: what I have to improve, what to do differently, etc.

I thought this document would fit well as that last menu option in the `Data Studio App` section, right after `Activity Log` – what do you think?

Thank you! :-)

File: docs/app/using-existing-tables.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+description:
+  Using Directus with existing tables has some requirements and restrictions. 
+readTime: 3 min read
+---
+
+# Using exisiting tables
+
+Directus recognizing tables only, when they adhere to vertain rules. Since different CMS systems handle SQL differently, it is recommended to first check how the foreign source structure its data. One main area of focus should be relations.
```
Comment (LasseRosenow) [inline docs/app/using-existing-tables.md:9]: "certain" => certain

**Comment by phazonoverload** [comment]
Thank you for your contribution, but we are not going to merge this PR. I don't think adding a whole page just to impart this note that's already in the docs elsewhere makes sense.

However, we will be sure to make this more explicit in other places in the docs where the audience may need to know.

**Comment by rowild** [comment]
@phazonoverload But this is still in progress. I am not done with it yet and it was set to "Draft". Why do you close that?

**Comment by phazonoverload** [comment]
I understand where you're coming from, but we didn't agree a scope of work and it's hard to know where you're going with this. I'd rather get this back on track with regards to our process than let you spend more time on something that isn't right. 

As mentioned here: https://docs.directus.io/contributing/community.html#education

**Comment by rowild** [comment]
OK, I understand to a certain extend. But I have never experienced that the team of a OSS project closes a commit that is set to "Draft". First asking would have been nice. Not sure why you put such a strange pressure on it, as if there was a deadline... It is also really strange to read sentences like "we will be more explicit in other places where the audience may need to know" – as if I wasn't one member of that audience... and looking in the wrong place. And what about this sentence: " let you spend more time on something that isn't right."? What is not right that I am doing here?

Anyway, I have some more written up and I think it could be helpful to those who want to use existing DBs. Maybe I provide another PR... but the motivation is low... (after all it is the 3rd time that I am pushed back even though the topic is important, in one case it was even the key to a bug...)

**Comment by phazonoverload** [comment]
I think I should have spent way more time writing my comment to make sure what I was thinking and feeling was clear - I'm sorry about that. Wanted to take the time this morning to reply properly. 

We have a process around accepting new docs contributions to make sure any work someone does actually writing stuff eventually results in a merge. The key is that an issue is opened first, we have a chat there about the scope and shape of contributions, and then we all know that it will eventually make it in, and that we commit time to working with an author to make the ebst document possible.

There's a whole host of reasons for this: 

1. **Is the overall scope and content agreed?** 
   There's no open issue, so it's unclear what will be contained in this document. Right now, it looks to just be about lack of support for composite primary keys. There's nothing that explains what else will be here. 
2. **Are we already working on it? Is this duplicated effort?** 
   It's already in the to do list, where "it" is making sure mentioning the lack of composite keys support is repeated in more places where folks are likely to spot it, and a dedicated guide on bringing your own DB. I would love to work on this with you as it remains a "to do", but it needs to start with that upfront convo in an issue, not a PR. 
3. **Does it belong in docs?**
   Less relevant for this because the answer is an enthusiastic "yes", but we often see docs suggestions/contributions that we don't believe should live in the docs. Docs are meant to be specific and curated, with the understanding that it becomes my team's job to maintain them. We'll soon have a developer blog open to contributions, where non-maintained and more specific/opinionated content can live. 
4. **Are examples you'll provide relevant and unopinionated?**
   We try really hard to provide unopinionated documentation to both allow for biggest impact and lowest maintenance burden. 

You also mention a few phrases I said as feeling a bit odd, and I wanted to give more context there as well. In trying to be brief, I fear they came off in a totally different way than intended.

* **"we will be more explicit in other places where the audience may need to know"**
  I simply meant that we'll take the notice already in the docs and make sure it's visible in more places where I think folks are more likely to be. For example, in our upcoming improvements to self-hosted guides. This is the quick win that requires little work and may have the required impact. 
* **"let you spend more time on something that isn't right.**
  Totally see how this can be read to be very hostile, and sorry for that. I simply meant that without an agreed scope of work and contribution in an issue, and even moreso with no detail in your PR description, I have no idea what you're planning. And I'd hate for anyone to jump in and spend time contributing for it to not make sense for any of the reasons above. 

I hope this explains my rationale a little more. I know our process can feel a bit long-winded, but it exists to both protect contributors' time and allow us to set time aside to properly collaborate with folks. 

I would still very much be interested in your contribution around bringing your own DB (in the self-hosted section), but I want to bump it back to an issue first. If that's of interest to you, open an issue and @ me in it, and we can go from there. 

If your motivation to contribute is still low, then please know that I'm sorry and I'll take this experience forward when working with other people who want to contribute - I believe the additional context upfront would have likely saved bad feeling. Have a nice day 😄

**Comment by rowild** [comment]
Thank you for clarifyng and taking the time! – Here is what happens: I do research to find out what the heck the mm-magic is that Directus need to recognise those relationships in existing DBs. To do that I need to setup at least 2 local instances of Directus with Docker. But this is a problem for me (I've never had to do that and DevOps stuff... well, you know: my main profession is musician, I am not a developer, so I do not have all that experience... but I am trying.) 

The thing is that I when I asked about help for this particular problem, the answers were very vage or even "That'S a Docker topic, ask there". Wow. Great. So I ask there. Guess what? "That's a Directus-specific problem. Ask there." ... So that's how to loose time and be stuck. At this point I don't even want to ask anymore anywhere. 

Things continue like that. I ask "what do I have to do to make Directus recognise mm tables? Or even a simple relation like user_created? How can I add such a field to my DB setup?" Answer (not only one): "You probably have to configure the collection." As if I haven't tried that already... so again no help, nevertheless I keep trying to find out what's happening.

All that takes time. Any now, compound keys are really not the only thing that should be mentioned when it comes to using existing DBs. BTW: the only feedback I got with my DRAFT was a grammar fix, nothing else... Question: do you not see what is missing here when it comes to using existing DBs? Sorry, if that sounds unfriendly, but there are Youtube videos (by Brian?) that say "Directus will do its best." – and that's about it what you get on info, when you want to use existing DBs.

What is lacking is a clear walk-through that explains what has to be done to setup proper relations. If you think that needs an issue first instead of right going into it and provide a DRAFT, well, ok. But ... man... it is already such a complicated, unsupported trip. You could have simply answered in the DRAFT, too, because the problem is there and it does not need a separate discussion – it needs a solution.

I have once more posted a question today on Directus after having failed all along. If I do not get any helpful answer to that, I leave the topic alone and will probably seek for another CMS, which is always sad, because it is a huge loss of time and more so because Directus is actually really great, but seems to struggle with the usual problem, and that is: those who manage it are way to deep into the matter to see what is needed for the basics. (That might be not true here in general, but in this particular case - using existing DBs and automatically building APIs on it, which is what Directus advertises – it is.)

(But just to give another example: Why not provide Docker setup guides for all DBs? After all you claim you want it to make simple. But it is not clear to everybody, that MariaDB still needs the DB_CLIENT params defined as "mysql", e.g. And it is not clear right away if I now need to change those MYSQL_USER, MYSQL_PASSWORD etc variables to MARIADB_USER, MARIADB_PASSWORD which would make sense, but doesn't work. Also, defining a prefix in the .env file must not have a dash, other wise Docker will not spin up.. well, ok, I know what you will say: those are DB and Docker related stuff. And you are right. But that is not the answer that helps making things simple. It would, however, be one of those things that would stand out compared to other documentation and make Directus even more awesome.)

**Comment by phazonoverload** [comment]
Not sounding unfriendly at all, and I think all very fair comments and stuff is totally missing. Improving the self-hosting experience through education is a key initiative of my team (especially at the starter level and not assuming folks have the deep knowledge). For context, my band of merry educators is fairly new still and we know there's work to do in all areas to make Directus as approachable as possible. 

Like I said, I'm happy to continue scoping out what a docs contribution might look like in an issue, which has to come before a PR is opened. This isn't to delay a solution, it's to make sure that the solution is the correct one before spending time on it.

**Pull request closed without merging**